core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorManagerService.js

```diff
@@ -138,6 +139,40 @@ oppia.factory('SimpleEditorManagerService', [
         data.questionList.addQuestion(QuestionObjectFactory.create(
           lastStateName, stateData.interaction, ''));
       },
+      changeQuestion: function(newQuestionType, index) {
+        var currentStateName = data.questionList.getAllStateNames()[index];
+        var nextStateName = data.questionList.getAllStateNames()[index + 1];
+        var questionCount = data.questionList.getQuestionCount();
+        var currentInteractionId = (
+          SimpleEditorShimService.currentInteractionId(currentStateName));
+        var doesLastQuestionHaveAnswerGroups = (
+          QuestionListObjectFactory.doesLastQuestionHaveAnswerGroups);
+
+        // Update Question Type If interactionId is not same.
+        if (newQuestionType !== currentInteractionId) {
+          var newAnswerGroups = [];
+          switch (newQuestionType) {
+            case 'MultipleChoiceInput':
+              SimpleEditorShimService.saveInteractionId(
+                currentStateName, DEFAULT_INTERACTION.ID);
+              SimpleEditorShimService.saveCustomizationArgs(
+                currentStateName, DEFAULT_INTERACTION.CUSTOMIZATION_ARGS);
+              newAnswerGroups.push(AnswerGroupObjectFactory.createNew([
+                RuleObjectFactory.createNew('Equals', {
+                  x: 0
+                })
+              ], OutcomeObjectFactory.createEmpty(nextStateName), false));
+              break;
+          }
+          if(doesLastQuestionHaveAnswerGroups && index !== questionCount - 1) {
```

Comment:
A last state gets added only when answers is selected, this to prevent the data getting corrupted when user add questions and then change it without selecting answer.